tests/test_sanity_check.py

```diff
@@ -85,22 +85,34 @@ def test_check_port_retries_negative(self):
             proc.terminate()
 
     def test_command_positive(self):
-        self.assertTrue(sanity_check.check_command("echo 1"))
+        command_passed, command_output = sanity_check.check_command("echo 1")
+        self.assertTrue(command_passed)
+        self.assertEqual(command_output, "1\n")
 
     def test_command_positive_and(self):
-        self.assertTrue(sanity_check.check_command("echo 1 && echo 2"))
+        command_passed, command_output = sanity_check.check_command("echo 1 && echo 2")
+        self.assertTrue(command_passed)
+        self.assertEqual(command_output, "1\n2\n")
 
     def test_command_positive_redirect(self):
-        self.assertTrue(sanity_check.check_command("echo foo canary bar | grep canary"))
+        command_passed, command_output = sanity_check.check_command("echo foo canary bar | grep canary")
+        self.assertTrue(command_passed)
+        self.assertEqual(command_output, 'foo canary bar\n')
 
     def test_command_negative(self):
-        self.assertFalse(sanity_check.check_command("exit 1"))
+        command_passed, command_output = sanity_check.check_command("exit 1")
+        self.assertFalse(command_passed)
+        self.assertEqual(command_output, '')
 
     def test_command_negative_and(self):
-        self.assertFalse(sanity_check.check_command("exit 1 && echo 1"))
+        command_passed, command_output = sanity_check.check_command("exit 1 && echo 1")
+        self.assertFalse(command_passed)
+        self.assertEqual(command_output, '')
 
     def test_command_negative_pipe(self):
-        self.assertFalse(sanity_check.check_command("echo foo bar | grep canary"))
+        command_passed, command_output = sanity_check.check_command("echo foo bar | grep canary")
+        self.assertFalse(command_passed)
+        self.assertEqual(command_output, '')
```

Comment:
@UmanShahzad Might be good to add a test scenario with exist code != 0 and some stdout/stderr output.

Reply:
@e-kolpakov If the command did not pass, that means it exited with code != 0. You can confirm this by reading the documentation about the `subprocess.check_output` function:

```
    If the exit code was non-zero it raises a CalledProcessError.  The
    CalledProcessError object will have the return code in the returncode
    attribute and output in the output attribute.
```

Reply:
@UmanShahzad that's not exactly what I meant. My point was that as far as I can tell, there are no test with `command_passed = false` and `command_output != ''`. Might make sense to add it to ensure that command stdout/stderr is actually captured and returned (especially stderr, since it doesn't seem to be covered by other existing tests)

Reply:
Ah, ok! I'll add a test for this.